plugins/catalog-backend-module-gitlab/src/providers/GitlabDiscoveryEntityProvider.ts

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { TaskRunner } from '@backstage/backend-tasks';
+import { Config } from '@backstage/config';
+import { GitLabIntegration, ScmIntegrations } from '@backstage/integration';
+import {
+  EntityProvider,
+  EntityProviderConnection,
+} from '@backstage/plugin-catalog-backend';
+import { LocationSpec } from '@backstage/plugin-catalog-backend';
+import { Logger } from 'winston';
+import {
+  GitLabClient,
+  GitLabProject,
+  GitlabProviderConfig,
+  paginated,
+  readGitlabConfigs,
+} from '../lib';
+import * as uuid from 'uuid';
+import { locationSpecToLocationEntity } from '@backstage/plugin-catalog-backend';
+
+type Result = {
+  scanned: number;
+  matches: GitLabProject[];
+};
+
+/**
+ * Extracts repositories out of an GitLab instance.
```

Comment:
More like "Discovers entity definition files in the groups of a GitLab instance" or something like that?